packages/sign-client/src/controllers/engine.ts

```diff
@@ -1388,7 +1393,11 @@ export class Engine extends IEngine {
       throw new Error(message);
     }
     const { topic, response } = params;
-    await this.isValidSessionTopic(topic);
+    // if the session is already disconnected, we can't respond to the request so we need to delete it
+    await this.isValidSessionTopic(topic).catch((error) => {
```

Comment:
Style nit: would prefer if we stick with imperative `try/catch + await` like in other places in the engine rather than mixing in Promise `then/catch`.

I get this is more convenient syntax wise but feels a bit out of place.